### PR TITLE
Add checks, fix bugs

### PR DESCRIPTION
**Pull request by slesaad**
Motivation:
Fixing issues: #38 #40 #41 #42 #43 #45 #47 #48 #50 #52 #57

TLDR;
Adds missing fields to various checks,
Reuses already existing checks for new checks,
Fixes URL Check not catching incomplete URL bug

### CODE REVIEW

File: pyQuARC/schemas/check_messages.json [hunk 1]
```diff
@@ -183,6 +183,22 @@
...
+        "remediation": "Update the start and/or end date time so that the timeline is logical."
+    },
+    "project_date_time_logic_check_2": {
+        "failure": "The Start Date Time `{}` and End Date Time `{}` are the same.",
+        "help": {
+            "message": "",
+            "url": ""
+        },
+        "remediation": "The temporal Start Date Time and End Date Time are identical."
+    },
```
Comment (CarsonDavis) [inline pyQuARC/schemas/check_messages.json:201]: Why does one remediate refer to the temporal dates but the other just refers to regular dates? Can we standardize the use of capitalization between the two messages?

File: pyQuARC/schemas/check_messages.json [hunk 2]
```diff
@@ -406,5 +422,29 @@
...
         },
         "remediation": "Make sure one of the fields is populated."
+    },
+    "spatial_coverage_type_check": {
+        "failure": "The Spatial Coverage Type provided `{}` is invalid.",
+        "help": {
+            "message": "",
+            "url": ""
+        },
+        "remediation": "Please provide a spatial coverage type from the following list: ['HORIZONTAL', 'VERTICAL', 'ORBITAL', 'HORIZONTAL_VERTICAL', ORBITAL_VERTICAL', 'HORIZONTAL_ORBITAL', 'HORIZONTAL_VERTICAL_ORBITAL']"
```
Comment (CarsonDavis) [inline pyQuARC/schemas/check_messages.json:432]: capitalization

Comment (slesaad) [reply pyQuARC/schemas/check_messages.json]: fixed in 72f7f19

File: pyQuARC/schemas/check_messages.json (hunk 1 above)
Comment (slesaad) [reply pyQuARC/schemas/check_messages.json]: fixed in 72f7f19

**Pull request merged**